docker/fedora-39.dockerfile

```diff
@@ -104,9 +140,43 @@ WORKDIR /build/sunshine/build
 RUN <<_MAKE
 #!/bin/bash
 set -e
+if [[ "${TARGETPLATFORM}" == "linux/amd64" ]]; then
+  export TUPLE=x86_64-linux-gnu
+elif [[ "${TARGETPLATFORM}" == "linux/arm64" ]]; then
+  export TUPLE=aarch64-linux-gnu
+else
+  echo "unsupported platform: ${TARGETPLATFORM}"
+  exit 1
+fi
+
+cat > toolchain.cmake <<_TOOLCHAIN
+set(CMAKE_ASM_COMPILER "${TUPLE}-gcc")
+set(CMAKE_ASM-ATT_COMPILER "${TUPLE}-gcc")
+set(CMAKE_C_COMPILER "${TUPLE}-gcc")
+set(CMAKE_CXX_COMPILER "${TUPLE}-g++")
+set(CMAKE_AR "${TUPLE}-gcc-ar" CACHE FILEPATH "Archive manager" FORCE)
+set(CMAKE_RANLIB "${TUPLE}-gcc-ranlib" CACHE FILEPATH "Archive index generator" FORCE)
+set(CMAKE_SYSTEM_PROCESSOR "${TARGETARCH}")
```

Comment:
Unfortunately, environment variables set in one `RUN` command don't make it into the later ones, and Dockerfiles aren't flexible enough for `ENV` values to be conditional. You need to repeat the code or write `TARGETARCH=foo` into a file and `source` that later.

Reply:
Oh... I missed that `TARGETARCH` was in this step.